lib/explorer/series.ex

```diff
@@ -2546,7 +2578,25 @@ defmodule Explorer.Series do
       >
   """
   @doc type: :element_wise
-  @spec add(left :: Series.t() | number(), right :: Series.t() | number()) :: Series.t()
+  @spec add(
+          left :: Series.t() | number() | NaiveDateTime.t(),
+          right :: Series.t() | number() | NaiveDateTime.t()
+        ) :: Series.t()
+  def add(%NaiveDateTime{} = left, %Series{dtype: {:duration, timeunit}} = right),
+    do: apply_series_list(:add, [left, right]) |> cast({:datetime, timeunit})
```

Comment:
Just a nitpick: I am not sure if we should cast here. If we cast here, we are assuming that the backend does not return something with the correct dtype. I think that's a backend logic: we say what is the expected dtype and the backend is responsible for dealing with it. If we remove the cast, what will happen? Do we get an integer or duration or datetime back?

If we get a different behaviour than polars, could this mean that we are actually not casting NaiveDateTime properly when sending it to Polars? Or are we choosing a different behaviour than Polars? If so, why?

The same rationale applies to `cast` in subtract, multiply, and divide.

Reply:
Right, I would like to avoid the postcast in general. I also think we should consider converting the NaiveDateTime to a series in `series.ex`, so the backend does not have to handle it either.

There was a reason why we didn't do it, but i don't fully recall. I will investigate.

Reply:
Will do.

Reply:
Did this for add and subtract here: [2f77e76](https://github.com/elixir-explorer/explorer/pull/683/commits/2f77e76d304bdc2bb76a4510fcdd542c35323247)

Multiply and divide seem determined to get returned as integers. Will investigate what's going on tomorrow.

Reply:
Ok I think I know what's going on. Polars seems to be treating scalar and element-wise multiplication differently.

https://www.rustexplorer.com/b/4cdukd

```rust

println!("{:?}", diff);                                  // [1d]
println!("{:?}", Series::new("prod1", &diff * 10));      // [10d]
println!("{:?}", Series::new("prod2", (10).mul(&diff))); // [10d]
println!("{:?}", Series::new("prod3", &ten * &diff));    // [864000000]
println!("{:?}", Series::new("prod4", &diff * &ten));    // [864000000]
```

Unfortunately it's the element-wise multiplication that's returning integers. Also:

```rust
println!("{:?}", Series::new("prod4", &diff * &diff));
//=> mul operation not supported for dtypes `duration[ms]` and `duration[ms]`
```

So I think we either have to drop support or post-cast.

I wonder if this is a bug in polars. Seems like an odd choice if it's intentional. Pandas seems to support what we're going for:

```python
import pandas as pd

duration = pd.Series([pd.Timedelta(1, "d"), pd.Timedelta(2, "d")])
integer = pd.Series([10, 10])
print(duration * integer)
#=> 0   10 days
#=> 0   20 days
#=> dtype: timedelta64[ns]
```